robot_skills/src/robot_skills/arms.py

```diff
@@ -22,42 +22,144 @@
 # Temporary: this should be approximately 0.02 once the sensor is correctly setup
 GRASP_SENSOR_LIMITS = tuple(rospy.get_param("skills/arm/grasp_sensor/limits", [0.0025, 0.18]))
 
+
+# Constants for arm requirements. Note that "don't care at all" is not here, as
+# it can be expressed by not imposing a requirement (set it to None).
+
+# Specific types of gripper.
+GRIPPER_TYPE_PINCH = "gripper-type-pinch"
+GRIPPER_TYPE_PARALLEL = "gripper-type-parallel"
+GRIPPER_TYPE_SUCTION = "gripper-type-suction"
+
+# Pseudo gripper types.
+GRIPPER_TYPE_GRASPING = "pseudo-gripper-type-any-grasping-will-do" # Either pinch or parallel
+GRIPPER_TYPE_NONE = "pseudo-gripper-type-no-gripper"
+
+# Pseudo objects for 'any object' or 'no object'.
+OBJECT_ANY  = "pseudo-object-saying-any-object-will-do"
+OBJECT_EMPTY = "pseudo-object-saying-lack-of-object-will-do"
+
+
 class PublicArm(object):
     """
     Public arm interface, also checks a challenge doesn't try to use more than it asked for.
 
     :ivar _arm: Private link to the real arm.
-    :ivar gripper_type: Gripper type to be used.
-    :ivar joint_goals: Joint goals to be used.
-    :ivar joint_trajectories: Joint trajectories to be used.
+    :vartype _arm: Arm
+
+    :ivar default_gripper_type: Gripper type to use if the user didn't provide one.
+    :vartpe default_gripper_type: str
+
+    :ivar _avail_gripper_types: Gripper types that may be used.
+    :vartype _avail_gripper_types: set of str (the GRIPPER_TYPES_* constants)
+
+    :ivar _has_occupied_by: Whether the arm supports 'occupied_by' calls.
+    :vartype _has_occupied_by: bool
+
+    :ivar _avail_joint_goals: Joint goals that may be used.
+    :vartype _avail_joint_goals: set of str
+
+    :ivar _avail_joint_trajectories: Joint trajectories that may be used.
+    :vartype _avail_joint_trajectories: set of str
     """
-    def __init__(self, arm, gripper_type, joint_goals, joint_trajectories):
+    def __init__(self, arm, avail_gripper_types, default_gripper_type,
+                 has_occupied_by, avail_joint_goals, avail_joint_trajectories):
         self._arm = arm
-        self.gripper_type = gripper_type
-        self.joint_goals = joint_goals
-        self.joint_trajectories = joint_trajectories
+        self.default_gripper_type = default_gripper_type
+        self._avail_gripper_types = avail_gripper_types
+        self._has_occupied_by = has_occupied_by
+        self._avail_joint_goals = avail_joint_goals
+        self._avail_joint_trajectories = avail_joint_trajectories
```

Comment:
I prefer clarify over saving those 4 'able' characters :-). Please use the full name